frontend/src/components/widgets/DataGrid/DataGrid.tsx

```diff
@@ -180,10 +180,47 @@ function DataGrid({
     onColumnResized,
   } = useDataLoader(element)
 
-  const [tableWidth, setTableWidth] = useState(propWidth)
+  const [width, setWidth] = useState(propWidth)
 
   const dataEditorRef = React.useRef<DataEditorRef>(null)
 
+  useLayoutEffect(() => {
+    // Without this timeout, the canvas does not reflect the actual size sometimes
+    setTimeout(() => {
+      // TODO(lukasmasuch): Support use_container_width parameter
+
+      let adjustedTableWidth = Math.min(
+        columns.length * MIN_COLUMN_WIDTH,
+        MIN_COLUMN_WIDTH
+      )
+
+      if (numRows) {
+        const firstCell = dataEditorRef.current?.getBounds(0, 0)
+        const lastCell = dataEditorRef.current?.getBounds(
+          columns.length - 1,
+          numRows - 1
+        )
+
+        if (firstCell && lastCell) {
+          // Calculate the table width, the +2 corresponds to the table borders
+          const fullTableWidth = lastCell.x - firstCell.x + lastCell.width + 2
+
+          // TODO(lukasmasuch): Also adjust the table height?
+          // const fullTableHeight = lastCell.y - firstCell.y + lastCell.height + 2
+
+          if (fullTableWidth < propWidth) {
+            adjustedTableWidth = fullTableWidth
+          } else {
+            // Use the max container width
+            adjustedTableWidth = propWidth
+          }
```

Comment:
nit: Could this be rewritten to use `Math.min`?

Reply:
ahh yes, of course 👍